packages/core/src/api/bot.ts

```diff
@@ -0,0 +1,101 @@
+import { makeURLSearchParams, type REST } from '@discordjs/rest';
+import {
+	Routes,
+	type APIGuildMember,
+	type APIDMChannel,
+	type APIPartialGuild,
+	type APIUser,
+	type RESTGetAPICurrentUserGuildsQuery,
+	type RESTPatchAPICurrentUserJSONBody,
+	type RESTPatchAPIGuildMemberJSONBody,
+	type RESTPatchAPIGuildVoiceStateCurrentMemberJSONBody,
+} from 'discord-api-types/v10';
+
+export class BotsAPI {
+	public constructor(private readonly rest: REST) {}
+
+	/**
+	 * Returns the user object of the requester's account
+	 */
+	public async getUser() {
+		return (await this.rest.get(Routes.user('@me'))) as APIUser;
+	}
```

Comment:
Can't you do it like this instead? this is more readable 
```suggestion
	public async getUser(): Promise<APIUser> {
		return this.rest.get(Routes.user('@me'));
	}
```

Reply:
That doesn't work it gives a type error:
```
Type 'unknown' is not assignable to type 'APIGuildMember'
```

Reply:
Personally, I'd prefer `return this.rest.get(Routes.user('@me')) as Promise<APIUser>;`. Looks more readable to me (and removes the unnecessary return await)